### PR TITLE
mpi/romio: Minor fixes for Lustre builds

### PR DESCRIPTION
**Pull request by tarudoodi**
## Pull Request Description
Minor fixes to support Lustre builds.
Uses MPL wrapper for memalign in romio/lustre.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by tarudoodi** [comment]
test:mpich/ch4/ofi

File: src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c [hunk 1]
```diff
@@ -4,6 +4,13 @@
  */
 
 #include "ad_lustre.h"
+#ifndef _XOPEN_SOURCE
+#define _XOPEN_SOURCE 500
+#endif
+#ifndef _ISOC11_SOURCE
+#define _ISOC11_SOURCE 1
+#endif
```
Comment (raffenet) [inline src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c:13]: a comment noting which features these defines enable would be helpful for someone reading the code

File: src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c [hunk 2]
```diff
@@ -101,7 +102,7 @@ static ssize_t ADIOI_LUSTRE_Directio(ADIO_File fd, const void *buf, MPI_Count le
                 return -1;
             nbytes += err;
         } else {
-            newbuf = (void *) memalign(LUSTRE_MEMALIGN, size);
+            newbuf = (void *) MPL_aligned_alloc(LUSTRE_MEMALIGN, size, MPL_MEM_OTHER);
```
Comment (raffenet) [inline src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c:105]: let's use `MPL_MEM_IO` for the class here

File: src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c [hunk 3]
```diff
@@ -123,7 +124,7 @@ static ssize_t ADIOI_LUSTRE_Directio(ADIO_File fd, const void *buf, MPI_Count le
...
```
Comment (raffenet) [inline src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c:127]: same comment about `MPL_MEM_IO`

File: src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c (hunk 1 above)
Comment (tarudoodi) [reply src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c]: Apparently this was only needed for our cluster. Removed it.

File: src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c (hunk 3 above)
Comment (tarudoodi) [reply src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c]: Fixed.

**Comment by tarudoodi** [comment]
test:mpich/ch4/ofi

**Review by raffenet — APPROVED** [review]

**Pull request merged**